src/codegen/generators/section_names.rs

```diff
@@ -40,6 +40,18 @@ pub fn mangle_type(index: &Index, ty: &typesystem::DataType) -> Result<section_m
                 Ok(acc)
             })?,
         },
+        DataTypeInformation::Array { inner_type_name, .. } => {
+            Type::Array { inner: Box::new(access_inner(inner_type_name)?) }
+        }
+        // FIXME: Is that correct?
```

Comment:
Yes, the range is only relevant at compile time AFAIK.
Technically the norm allows for different runtime behaviour if a value is out of range, but this would still not affect the generated type, only some functions that the user needs to have. 
The subrange has an implicit method that gives the upper and lower bounds but this is resolved on compile time as well since the range is known.